src/coreclr/nativeaot/System.Private.CoreLib/src/System/ValueType.cs

```diff
@@ -164,7 +164,8 @@ private unsafe int RegularGetValueTypeHashCode(ref byte data, int numFields)
                     var fieldValue = (ValueType)RuntimeImports.RhBox(fieldType, ref fieldData);
                     if (fieldValue != null)
                     {
-                        hashCode = fieldValue.GetHashCodeImpl();
+                        // call virtual method to handle overriden case
+                        hashCode = fieldValue.GetHashCode();
```

Comment:
I think `GetHashCodeImpl` can now be inlined into `GetHashCode`.

The only reason we had `GetHashCodeImpl` was because it was emulating CoreCLR behavior that _did not_ call the `GetHashCode` on the field type (even if it had one) but instead used the fallback algorithm.

Reply:
Is it feasible to use `System.HashCode` for NativeAOT here? Will it bring unintended dependency or complexity?

Reply:
I think it would be fine to use `System.HashCode`. This was mirroring the CoreCLR-JIT logic but now it diverged because that's how forks usually end up.

If we really wanted to, we could probably unify this to the point of sharing a part of this file, maybe with an ifdef or two. From a quick look `GetHashCodeStrategy` could be implemented using NativeAOT's `__GetFieldHelper`.